Samples/CayenneDecoder/Classes/CayenneDecoder.cs

```diff
@@ -0,0 +1,227 @@
+        {
+            var ret = GetOutValues(CayenneTypes.Accelerator);
+            if (ret != null)
+            {
+                accelerator = new Accelerator()
+                {
+                    Channel = ret[0],
+                    X = ((Int16)(ret[1] << 8) + ret[2]) / 1000.0,
+                    Y = ((Int16)(ret[3] << 8) + ret[4]) / 1000.0,
+                    Z = ((Int16)(ret[5] << 8) + ret[6]) / 1000.0
+                };
+                return true;
+            }
+            accelerator = null;
+            return false;
+        }
+
+        public bool TryGetBarometer(out Barometer barometer)
+        {
+            var ret = GetOutValues(CayenneTypes.Barometer);
+            if (ret != null)
+            {
+                barometer = new Barometer() { Channel = ret[0], Value = ((ret[1] << 8) + ret[2]) / 10.0 };
+                return true;
+            }
+            barometer = null;
+            return false;
+        }
+
+        public bool TryGetGyrometer(out Gyrometer gyrometer)
+        {
+            var ret = GetOutValues(CayenneTypes.Gyrometer);
+            if (ret != null)
+            {
+                gyrometer = new Gyrometer()
+                {
+                    Channel = ret[0],
+                    X = ((Int16)(ret[1] << 8) + ret[2]) / 100.0,
+                    Y = ((Int16)(ret[3] << 8) + ret[4]) / 100.0,
+                    Z = ((Int16)(ret[5] << 8) + ret[6]) / 100.0
+                };
+                return true;
+            }
+            gyrometer = null;
+            return false;
+        }
+
+        public bool TryGetGPSLocation(out GPSLocation gPSLocation)
+        {
+            var ret = GetOutValues(CayenneTypes.Gps);
+            if (ret != null)
+            {
+                gPSLocation = new GPSLocation();
+
+                gPSLocation.Channel = ret[0];
+                double sign = 1.0;
+                if ((ret[1] & 0x80) == 0x80)
+                {
+                    ret[1] = (byte)(ret[1] & 0x7F);
+                    sign = -1.0;
+                }
+                gPSLocation.Latitude = sign * ((ret[1] << 16) + (ret[2] << 8) + ret[3]) / 10000.0;
+                sign = 1.0;
+                if ((ret[4] & 0x80) == 0x80)
+                {
+                    ret[4] = (byte)(ret[4] & 0x7F);
+                    sign = -1.0;
+                }
+                gPSLocation.Longitude = sign * ((ret[4] << 16) + (ret[5] << 8) + ret[6]) / 10000.0;
+                sign = 1.0;
+                if ((ret[7] & 0x80) == 0x80)
+                {
+                    ret[7] = (byte)(ret[7] & 0x7F);
+                    sign = -1.0;
+                }
+                gPSLocation.Altitude = sign * ((ret[7] << 16) + (ret[8] << 8) + ret[9]) / 100.0;
+                return true;
+            }
+            gPSLocation = null;
+            return false;
+        }
+
+        private byte[] GetOutValues(CayenneTypes cayenneTypes)
```

Comment:
Could return a ReadOnlySpan<byte> here, not need to make a copy of byte[]

Reply:
I'm not keeping the original version of the array. I'm removing data in the original one and extract part of it. So I have to make a copy of it.